system/xmllib/xpathprocessor.hpp

```diff
@@ -167,8 +173,9 @@ interface IEsdlScriptContext : extends IInterface
     virtual void cleanupBetweenScripts() = 0;
     virtual void setTraceToStdout(bool val) = 0; //keep it simple for now.  default is to use jlog.  This flag may go away when we give more control over tracing
     virtual bool getTraceToStdout() = 0;
+    virtual IInterface *queryFunctionRegister() = 0;
 };
 
-extern "C" XMLLIB_API IEsdlScriptContext *createEsdlScriptContext(void * espContext);
+extern "C" XMLLIB_API IEsdlScriptContext *createEsdlScriptContext(void * espContext, IInterface *functionRegister);
```

Comment:
Should this be a IEsdlFunctionRegister?

Reply:
The use of IInterface is because IEsdlFunctionRegister is declared within the ESP and not appropriate to be referenced in system. Same reason espContext is void*. If the final comment from my review is valid, this new pointer will be moved to an ESP-specific interface and will be typed correctly.